### PR TITLE
Patch Z3 to support third party memory allocators

### PR DESCRIPTION
**Pull request by remysucre**
Workaround for https://github.com/ahumenberger/Z3.jl/issues/28

### CODE REVIEW

File: Z/z3/build_tarballs.jl [hunk 1]
```diff
@@ -36,9 +35,12 @@ fi
 """
 
 # Bash recipe for building across all platforms
+# Patches Z3 to work around https://github.com/Z3Prover/z3/issues/6890 (closed but not fixed!)
```
Comment (imciner2) [inline Z/z3/build_tarballs.jl:38]: This is better placed right above the line doing the patching, and also a description of what the patch does - e.g. why is that sed command patching it? The linked issue doesn't mention this as a fix at all.

**Review by imciner2 — DISMISSED** [review]

File: Z/z3/build_tarballs.jl (hunk 1 above)
Comment (fingolfin) [inline Z/z3/build_tarballs.jl:38]: If this comment refers to the `sed` invocation below, why not move it right next to that?

File: Z/z3/build_tarballs.jl [hunk 2]
```diff
@@ -9,7 +9,6 @@ delete!(Pkg.Types.get_last_stdlibs(v"1.6.3"), uuid)
 
 name = "z3"
 version = v"4.13.0"
-julia_versions = [v"1.6.3", v"1.7", v"1.8", v"1.9", v"1.10", v"1.11"]
```
Comment (fingolfin) [inline Z/z3/build_tarballs.jl:12]: Thanks :-)

File: Z/z3/build_tarballs.jl [hunk 3]
```diff
@@ -36,9 +35,12 @@ fi
 """
 
 # Bash recipe for building across all platforms
+# Patches Z3 to work around https://github.com/Z3Prover/z3/issues/6890 (closed but not fixed!)
 script = macfix * raw"""
 cd $WORKSPACE/srcdir/z3-*/core
 
+sed -i '16,30d' src/util/memory_manager.cpp
```
Comment (fingolfin) [inline Z/z3/build_tarballs.jl:42]: That's rather opaque and brittle, isn't it? If someone else wants to update the recipe in the future, they'll first have to reverse engineer what exactly this does.

Can't you instead e.g. specify the exact string to be deleted? Or provide an actual patchfile?

File: Z/z3/build_tarballs.jl (hunk 1 above)
Comment (remysucre) [reply Z/z3/build_tarballs.jl]: I linked to a new issue that documents more details about the problem, and moved the comment down.

File: Z/z3/build_tarballs.jl (hunk 3 above)
Comment (remysucre) [reply Z/z3/build_tarballs.jl]: Agreed, pushed a more explicit command and linked to a new issue documenting the bug.

**Pull request merged**